pkg/apis/wksprovider/controller/wksctl/machine_controller.go

```diff
@@ -1153,39 +1073,52 @@ func (a *MachineActuator) recordEvent(object runtime.Object, eventType, reason,
 	}
 }
 
-func getMachineID(machine *clusterv1.Machine) string {
+func getMachineID(machine *baremetalspecv1.BareMetalMachine) string {
 	return machine.Namespace + ":" + machine.Name
 }
 
-func (a *MachineActuator) getMachineAddress(machine *clusterv1.Machine) string {
-	m, _ := a.codec.MachineProviderFromProviderSpec(machine.Spec.ProviderSpec)
+func (a *MachineController) getMachineAddress(m *baremetalspecv1.BareMetalMachine) string {
+	if m.Spec.Private.Address != "" {
+		return m.Spec.Private.Address
+	}
+	return machineIPs[getMachineID(m)]
+}
 
-	if m.Private.Address != "" {
-		return m.Private.Address
+func (a *MachineController) SetupWithManager(mgr ctrl.Manager, options controller.Options) error {
+	controller, err := ctrl.NewControllerManagedBy(mgr).
+		WithOptions(options).
+		For(&baremetalspecv1.BareMetalMachine{}).
+		Watches(
+			&source.Kind{Type: &clusterv1.Machine{}},
+			&handler.EnqueueRequestsFromMapFunc{
+				ToRequests: util.MachineToInfrastructureMapFunc(baremetalspecv1.SchemeGroupVersion.WithKind("BareMetalMachine")),
```

Comment:
There should IMO be a constant for this string, or even a pre-made GVK. I can do this later though.